assets/src/components/app.tsx

```diff
@@ -48,23 +48,24 @@ const AppRoutes = () => {
       </div>
       <VehiclesByRouteIdProvider vehiclesByRouteId={vehiclesByRouteId}>
         <div className="m-app__main">
-          <TabBar
+          <Nav
             pickerContainerIsVisible={pickerContainerIsVisible}
             openView={openView}
-          />
-          <BrowserRoute exact={true} path="/" component={LadderPage} />
-          <BrowserRoute
-            exact={true}
-            path="/shuttle-map"
-            component={ShuttleMapPage}
-          />
-          <BrowserRoute
-            exact={true}
-            path="/settings"
-            component={SettingsPage}
-          />
-          <BrowserRoute exact={true} path="/search" component={SearchPage} />
-          {openView === OpenView.Late ? <LateView /> : null}
+          >
+            <BrowserRoute exact={true} path="/" component={LadderPage} />
```

Comment:
React Router v6 [(docs)](https://reactrouter.com/docs/en/v6/upgrading/v5) has a whole bunch of significant changes to the way routing is done. Have you considered whether/when you want to upgrade, and how those changes might interact with this PR, or how they might affect your plans for how the code will be refactored?

This approach of putting nav around the routes works for v5, but it might be incompatible with v6.

As an example, here's Glides' upgrade, and it turned out pretty messy and significant: https://github.com/mbta/glides/pull/529

This might be easier to talk about synchronously.

Reply:
Hmm, I hadn't been keeping up with React Router so I was unaware that there was a new version with potentially-breaking changes. Let me dig into the docs and PR you linked to a bit and see if it's obvious whether or not this approach will work with 6.

Reply:
Okay, so after reviewing the [upgrade guide](https://reactrouter.com/docs/en/v6/upgrading/v5) I think that upgrading to version 6 will require some minor changes in a few places, but won't require any major refactors. It looks like Glides' use of React Router is somewhat more complicated than Skate's already, so there are really only a couple of basic components and features that we're using.

Reply:
Yeah, Glides parses path params, and I guess Skate doesn't anywhere.

I'm particularly worried about `Route` elements having to be direct children of `Routes` elements, while this PR takes a step away from that by nesting `Route` inside `Nav`, but that's a problem that will be totally solvable after this navigation redesign.

Reply:
Ideally I'd like to continue having a separate `<Nav>` component even once it's no longer doing the work of gating the beta group. Do you see anything specific in the release notes about `<Route>` elements needing to be direct children? It works in 5 so I would assume they would call it out if it doesn't work in 6 anymore. In any event I could always go in the other direction (move all of the router / routes stuff to the inside of the `<Nav>`), so I think that one way or another I have a path forward.

Reply:
[Here's the details.](https://gist.github.com/mjackson/d54b40a094277b7afdd6b81f51a0393f)

Oh, it only applies with `<Switch>` (v5) or `<Routes>` (v6) which you aren't using. I didn't realize you could have a `<Route>` without a `<Switch>`, but even if you had to add `<Routes>` during your upgrade to v6, you could add it inside the `<Nav>`, so this won't end up being an issue.

I believe the `<BrowserRouter>` can be far up the tree, so that's okay, too.

False alarm.

Reply:
Okay, whew!

In any event thanks for the heads up, this got me looking at the upgrade guide and thinking about it.